src/viam/sdk/components/board/client.cpp

```diff
@@ -0,0 +1,226 @@
+#include <viam/sdk/components/board/client.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/component/board/v1/board.grpc.pb.h>
+
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/components/board/board.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/robot/client.hpp>
+
+namespace viam {
+namespace sdk {
+
+BoardClient::BoardClient(std::string name, std::shared_ptr<grpc::Channel> channel)
+    : Board(std::move(name)),
+      stub_(viam::component::board::v1::BoardService::NewStub(channel)),
+      channel_(std::move(channel)){};
+
+Board::status BoardClient::get_status() {
+    viam::component::board::v1::StatusRequest request;
+    viam::component::board::v1::StatusResponse response;
+
+    grpc::ClientContext ctx;
+
+    *request.mutable_name() = this->name();
```

Comment:
Because, for reasons I dont fully understand, `protoc` does not generate a `set_name` function. There are a couple other members like this (like `duration`) where there is no setter.